python/ray/dashboard/modules/job/job_manager.py

```diff
@@ -537,8 +549,9 @@ async def submit_job(
                 resources=entrypoint_resources,
                 scheduling_strategy=scheduling_strategy,
                 runtime_env=self._get_supervisor_runtime_env(
-                    runtime_env, submission_id, resources_specified
+                    runtime_env, submission_id, resources_specified, virtual_cluster_id
                 ),
+                _metadata=virtual_cluster_metadata,
```

Comment:
The `_metadata` is an extended options for Ray libraries, I'm not sure if it's ok to assign `virtual_cluster_metadata` to `_metadata` directly.

Reply:
I see, we should implement a scheduling strategy to achieve virtual cluster affinity scheduling